packages/ag-charts-types/src/chart/toolbarOptions.ts

```diff
@@ -42,10 +39,21 @@ export interface AgToolbarButton {
     ariaLabel?: string;
     /** Tooltip text to display on hover over the button. */
     tooltip?: string;
+}
+
+export interface AgToolbarButton extends AgToolbarButtonConfig {
+    /** Section name used for grouping of buttons.
+     *
+     * Adjacent buttons with the same section are grouped together.*/
+    section?: string;
     /** Value provided to caller when the button is pressed. */
     value: any;
     /** ID of the button (must be set when value is not a primitive) */
     id?: string;
+    /** The button type (default: `'button'`); `'switch'` is the same but also have an on/off state like checkboxes. */
+    role?: 'button' | 'switch';
+    /** Overrides for the button when checked. Only applicable with `role: 'switch'`. */
+    checkedOverrides?: AgToolbarButtonConfig;
```

Comment:
If `checkedOverrides` is only applicable when `role === 'switch'`, if it is feasible it would be good to bake it into the types, depending on how complicated it makes things. Don't worry if not.

```
interface AgToolbarButton {
    role?: 'button';
    // ...
}

interface AgToolbarSwitch {
    role: 'switch';
    checkedOverrides?: AgToolbarButtonConfig
    // ...
}

interface AgToolbarAnnotationOptionsGroup extends AgToolbarGroup {
    buttons?: Array<AgToolbarAnnotationOptionsButton | AgToolbarAnnotationOptionsSwitch>;
}

interface AgToolbarAnnotationOptionsSwitch extends AgToolbarSwitch {
    value: AgToolbarAnnotationOptionsButtonValue;
}
```

Reply:
Yes, I thought of that too but didn't go through with it. I think it's a good idea though, I'll see whether or not it's complicated.

Reply:
Done.